File: frontend/lib/screens/Employee/Employee_/post.dart
class Post {
  String id;
  String poster_id;
  int number;
  String description;
  String category;
  String? image;

  // ---- Add the post owner's data when needed ---- //

  String? posterName;
  String? posterProfilePicture;

  // ------------------------------------------------ //

  Post({
    required this.poster_id,
    required this.id,
    required this.number,
    required this.description,
    required this.category,
    this.image,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['_id'],
      number: json['number'],
      description: json['description'],
      category: json['category'],
      image: json['image'],
      poster_id: json['poster_id'],
    );
  }
}

File: backend/models/Post.js
const mongoose = require("mongoose");
const Joi = require("@hapi/joi");

const postSchema = mongoose.Schema({
  poster_id: {
    type: mongoose.ObjectId,
    required: true,
  },
  posterName: {
    type: String,
  },
  description: {
    type: String,
    required: true,
    min: 5,
    max: 200,
  },
  category: {
    type: String,
    required: true,
  },
  number: {
    type: Number,
    required: true,
    min: 1,
  },
  createdAt: {
    type: Date,
    required: true,
    default: Date.now,
  },
  image: {
    type: String,
  },
});

const schema = Joi.object({
  poster_id: Joi.string().hex().length(24).required(),
  description: Joi.string().min(5).max(200).required(),
  category: Joi.string().required(),
  number: Joi.number().min(1).required(),
});

module.exports = {
  postValidation: (data) => schema.validate(data),
  Post: mongoose.model("Post", postSchema),
};
